src/main/java/com/ning/compress/lzf/impl/UnsafeChunkEncoderBE.java

```diff
@@ -29,6 +29,11 @@ public UnsafeChunkEncoderBE(int totalLength, BufferRecycler bufferRecycler, bool
     @Override
     protected int tryCompress(byte[] in, int inPos, int inEnd, byte[] out, int outPos)
     {
+        // Sanity checks; otherwise if any of the arguments are invalid `Unsafe` might corrupt memory
+        _checkArrayIndices(in, inPos, inEnd);
+        _checkArrayIndices(out, outPos, out.length);
+        // TODO: Validate that `out.length - outPos` is large enough?
```

Comment:
I think this has to be validated because it does not look like the code below performs any bounds checks, but I am not sure what the correct minimum size here is. When I used `LZFChunk.MAX_CHUNK_LEN`[^1], a few unit tests were failing, which were using smaller output buffers.

Or alternatively it would be necessary to add bounds checks below to make sure `Unsafe` does not write outside `out`.

[^1]: Maybe the minimum size would have to be even larger, including `LZFChunk.MAX_HEADER_LEN`?